eiger_io/fs_handler.py

```diff
@@ -20,3 +21,53 @@ def __call__(self, seq_id):
         master_path = '{}_{}_master.h5'.format(self._base_path, seq_id)
         # TODO Return a multi-dimensional PIMS seq.
         return np.array(EigerImages(master_path))
+
+
+class FixedEigerImages(EigerImages):
+    def __init__(self, path, metadata):
+        super().__init__(path)
+        self._metadata = metadata
+
+    @property
+    def md(self):
+        return self._metadata
+
+    @property
+    def dtype(self):
+        return self.pixel_type
+
+    @property
+    def shape(self):
+        return self.frame_shape
+
+
+class LazyEigerHandler(HandlerBase):
+    specs = {'AD_EIGER'} | HandlerBase.specs
+    def __init__(self, fpath, frame_per_point, mapping=None):
+        # create pims handler
+        self.vals_dict = EIGER_MD_DICT.copy()
+        if mapping is not None:
+            self.vals_dict.update(mapping)
+        self._base_path = fpath
+        self.fpp = frame_per_point
+
+    def __call__(self, seq_id):
+        master_path = '{}_{}_master.h5'.format(self._base_path, seq_id)
+        md = {}
+        print('hdf5 path = %s' % master_path)
+        with h5py.File(master_path, 'r') as f:
+            md = {k: f[v].value for k, v in self.vals_dict.items()}
+        # the pixel mask from the eiger contains:
+        # 1  -- gap
+        # 2  -- dead
+        # 4  -- under-responsive
+        # 8  -- over-responsive
+        # 16 -- noisy
+        pixel_mask = md['pixel_mask']
```

Comment:
We've all been there, haha. That numpy implementation looks good to me -- maybe no need to cast to int though, depending on downstream use.

I would suggest putting this in a separate field, maybe `md['binary_mask']` so that the detailed information from Eiger is still accessible.